components/api/core/api_impl.bal

```diff
@@ -58,35 +59,117 @@ public function getTokens (http:Request getTokensReq) returns http:Response {
     }
 }
 
+# Search based on organization name
+#
+# + listOrgsReq - Received query parameters
+# + return - http response which cater to the request
+public function listOrgs(http:Request listOrgsReq) returns http:Response {
+    if (listOrgsReq.hasHeader(constants:ORG_NAME)) {      
+        if (listOrgsReq.hasHeader(constants:RESULT_LIMIT) && listOrgsReq.hasHeader(constants:OFFSET)) {
```

Comment:
According to this logic offset and limit seems mandatory. Shall we make the optional ? In case if those are not present offset will be 0 and result limit will be like 10. Also we need to enforce upper limit to resultLimit. We should not allow to request for limit 100 or more. Having 50 seems reasonable.

Reply:
Also usually result limit and offset are sent as query params.